lib/util/http/index.ts

```diff
@@ -172,6 +172,23 @@ export class Http<GetOptions = HttpOptions, PostOptions = HttpPostOptions> {
     return this.request<string>(url, { ...options, method: 'head' });
   }
 
+  protected requestBuffer<T>(
+    url: string | URL,
+    httpOptions?: InternalHttpOptions
+  ): Promise<HttpResponse<T> | null> {
+    return this.request<T>(url, {
+      ...httpOptions,
+      responseType: 'buffer' as any,
+    });
+  }
```

Comment:
```suggestion
  protected requestBuffer<Buffer>(
    url: string | URL,
    httpOptions?: InternalHttpOptions
  ): Promise<HttpResponse<Buffer> | null> {
    return this.request<Buffer>(url, {
      ...httpOptions,
      responseType: 'buffer',
    });
  }
```

It returns a buffer, so should not be generic.

Update `InternalHttpOptions.responseType` to accept `buffer` instead of any cast.

Reply:
> It returns a buffer, so should not be generic.

It was apparently a mistake, thanks!

> Update `InternalHttpOptions.responseType` to accept `buffer` instead of any cast.

This is harder than it looks, since `got` defines overloads depending on `responseType` and we cannot transparently pass through overloads in TypeScript. Anyway, I'll try my best for a better type-safety.

Reply:
7340cbe, 0db6f34